examples/jsm/loaders/GLTFLoader.js

```diff
@@ -4046,58 +4087,13 @@ function buildNodeHierarchy( nodeId, parentObject, json, parser ) {
 
 		// build skeleton here as well
 
-		let skinEntry;
-
-		return parser.getDependency( 'skin', nodeDef.skin ).then( function ( skin ) {
-
-			skinEntry = skin;
-
-			const pendingJoints = [];
-
-			for ( let i = 0, il = skinEntry.joints.length; i < il; i ++ ) {
-
-				pendingJoints.push( parser.getDependency( 'node', skinEntry.joints[ i ] ) );
-
-			}
-
-			return Promise.all( pendingJoints );
-
-		} ).then( function ( jointNodes ) {
+		return parser.getDependency( 'skin', nodeDef.skin ).then( function ( skeleton ) {
 
 			node.traverse( function ( mesh ) {
 
-				if ( ! mesh.isMesh ) return;
-
-				const bones = [];
-				const boneInverses = [];
-
-				for ( let j = 0, jl = jointNodes.length; j < jl; j ++ ) {
-
-					const jointNode = jointNodes[ j ];
-
-					if ( jointNode ) {
-
-						bones.push( jointNode );
-
-						const mat = new Matrix4();
-
-						if ( skinEntry.inverseBindMatrices !== undefined ) {
-
-							mat.fromArray( skinEntry.inverseBindMatrices.array, j * 16 );
-
-						}
-
-						boneInverses.push( mat );
-
-					} else {
-
-						console.warn( 'THREE.GLTFLoader: Joint "%s" could not be found.', skinEntry.joints[ j ] );
-
-					}
-
-				}
+				if ( ! mesh.isSkinnedMesh ) return;
 
-				mesh.bind( new Skeleton( bones, boneInverses ), mesh.matrixWorld );
+				mesh.bind( skeleton, mesh.matrixWorld );
```

Comment:
For reviewers: If we want to keep making different `Skeleton` instance for each `SkinnedMesh`, we may clone `skeleton`. Any strong reasons why we want to make different skeleotn instances?

Reply:
Think about two skinned meshes with a shared skeleton. IIRC you only need cloned skeletons with cloned bones when you want to animate both skinned meshes _independently_. Different skeleton instances but with shared bones would not be sufficient here. Since the previous code did not clone bones, this new approach seems to be cleaner.

Reply:
I read through the glTF spec

https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html

If I understand correctly, to match the spec the multiple skinned meshes referring to the same skin (skeleton) shouldn't animate independently because

* Multiple nodes can refer to the same skin (skeleton)
* A skeleton consists of joint nodes which must be in a scene graph
* Key frame skinning animation is achieved by animating joint nodes

So, reusing the same skeleton instance without cloning matches the spec, I think?

Reply:
Do you happen to have an example file for this? Happy to provide one otherwise.

Reply:
@donmccurdy Are you okay with the assumptions made in this PR? It would be great to include this nice refactoring in `r147`.

Reply:
Yes, this sounds good to me – thanks!